packages/teams-js/src/public/pages.ts

```diff
@@ -216,27 +251,16 @@ export namespace pages {
       sendMessageToParent('settings.setValidityState', [validityState]);
     }
 
-    /**
-     * Gets the config for the current instance.
-     * @returns Promise that resolves with the {@link Config} object.
-     */
-    export function getConfig(): Promise<Config> {
-      return new Promise<Config>(resolve => {
-        ensureInitialized(FrameContexts.content, FrameContexts.settings, FrameContexts.remove, FrameContexts.sidePanel);
-        resolve(sendAndUnwrap('settings.getSettings'));
-      });
-    }
-
     /**
      * Sets the config for the current instance.
      * This is an asynchronous operation; calls to getConfig are not guaranteed to reflect the changed state.
-     * @param Config The desired config for this instance.
+     * @param instanceConfig The desired config for this instance.
      * @returns Promise that resolves when the operation has completed.
      */
-    export function setConfig(instanceSettings: Config): Promise<void> {
+    export function setConfig(instanceConfig: InstanceConfig): Promise<void> {
```

Comment:
>setConfig [](http://example.com/codeflow?start=20&length=9)

Why does this one stay here when getConfig moved into pages?

Reply:
`setConfig` is only (afaik) applicable in `FrameContext.settings` wheras `getConfig` is available in all the FrameContexts. If we were to move `setConfig` to top-level `pages` it would require the developer to do both `pages.isSupported` and `context.pages.FrameContext === FrameContexts.settings` (or whatever the property on context is that has frameContext on it). I feel like leaving it in `pages.config` (which after moving `getConfig` out of only has methods available in `frameContexts.settings` and `frameContexts.remove` makes it easier for the developer to reason about what is and isn't available.

Reply:
I would expect us to have to have a deeper conversation on how `isSupported` on capabilities play with `frameContext` at some point, but with how things work right now I think leaving `setConfig` in `pages.config` gives a better dev experience to the app developer.